src/Courses/TopicGrades/GradeInfoHeader.tsx

```diff
@@ -42,154 +41,260 @@ const useStyles = makeStyles((theme) => ({
         }
 
         const params = {
-            topicId: topicId,
-            userId: grade.userId
+            topicId: topic.id,
+            userId: selected.user?.id // check selected.user not nil?
         };
         const result = await getGrades(params);
         setTopicGrade(result.data.data.first?.average ?? null);
     };
     useEffect(() => {
+        logger.debug('GradeInfoHeader: there has been a change in user or topic, fetching new topic grade.');
         fetchTopicGrade();
-    }, [grade.userId, topicId]);
+    }, [selected.user, topic]);
 
-    const workbookList = (vMap: Record<number, Array<number>>, versioned: boolean) => {
-        const options: WorkbookOption[] = [];
-        const versionKeys = _.keys(vMap).map((v) => { return parseInt(v, 10); }).sort();
+    const onSuccess = (gradeOverride: Partial<StudentGrade>) => {
+        logger.debug('GradeInfoHeader: overriding grade', gradeOverride.effectiveScore);
+        const currentGrade = grade; 
+        if (!_.isNil(gradeOverride) &&
+            !_.isNil(gradeOverride.effectiveScore) &&
+            !_.isNil(currentGrade)
+        ) {
+            currentGrade.effectiveScore = gradeOverride.effectiveScore;
+            setGrade(currentGrade);
+        }
+    };
+
+    const versionList = (vMap: Record<number, Array<number>>) => {
+        const versions: WorkbookOption[] = [];
+        const versionKeys = _.keys(vMap).map((v) =>  { return parseInt(v, 10); }).sort();
         versionKeys.forEach((key, i) => {
-            let prefix = '';
-            if (versioned) prefix = `Version #${(i + 1)} - `;
-            vMap[key].sort().forEach((v, i) => {
-                const value = v;
-                const label = prefix + `Attempt #${(i + 1)}`;
-                options.push({ label, value });
-            });
+            versions.push({ label: `Version #${i+1}`, value: key});
         });
-        return options;
+        return versions;
     };
 
+    const versionSubList = (vMap: Record<number, Array<number>>, versionKey: number) => {
+        const attempts: WorkbookOption[] = [{ label: 'current', value: -1 }];
+        vMap[versionKey].sort().forEach((id, index) => {
+            attempts.push({ label: `Attempt #${index+1}`, value: id});
+        });
+        return attempts;
+    };
+
+    // const workbookList = (vMap: Record<number, Array<number>>, versioned: boolean) => {
+    //     const options: WorkbookOption[] = [];
+    //     const versionKeys = _.keys(vMap).map((v) => { return parseInt(v, 10); }).sort();
+    //     versionKeys.forEach((key, i) => {
+    //         let prefix = '';
+    //         if (versioned) prefix = `Version #${(i + 1)} - `;
+    //         vMap[key].sort().forEach((v, i) => {
+    //             const value = v;
+    //             const label = prefix + `Attempt #${(i + 1)}`;
+    //             options.push({ label, value });
+    //         });
+    //     });
+    //     return options;
+    // };
+
     type Props = {
```

Comment:
This is old but we should follow our naming convention for props, that way we can move it to a separate file if we want or export it so that the type can be used elsewhere with the name being mindful